### PR TITLE
Added PlaceBuildingInit.

### PR DESCRIPTION
**Pull request by IceReaper**
The basical idea is to know how an actor is build / created. Not just by building it from the ui. I only added this one case, as its the only one i need currently. A crate may spawn an actor. A factory may produce an actor. A building may be placed by the player. It might spawn cause its some kind of reinforcement. The basical idea is to have an actor which may behave differently under certain circumstances, without this feature, the traits will keep triggering equaly.

Example A: Imagine we have an oil tanker vehicle. When it gets produced by a factory, it should wait for its assignment. However, in case B we have an oil derrick, and the vehicle is granted by FreeActor. It should search for the next derrick on spawn and start driving towards it. (kknd logic)

Example B: I have buildings which will build-up themself after placement. However, this would be the general case, all buildings which are spawned by script, unit deployment, or whatever other reason will start to build at this point, consuming money.

### CODE REVIEW

**Comment by pchote** [comment]
This feels speculatively general to me - my gut feeling is that this isn't going to get much use outside of your own use case.

I think a better approach may be to define a new `IProducedActorInitModifier` interface, similar to the existing `IDeathActorInitModifier`, and then call that from `PlaceBuilding`, `Production` and `ClonesProducedUnits`.  This would allow arbitrary other traits to add arbitrary other info, which would be a lot more flexible.

**Comment by IceReaper** [comment]
Sounds good, will do!

**Comment by pchote** [comment]
Any progress here @IceReaper?

**Comment by IceReaper** [comment]
Updated!

**Review by pchote — COMMENTED** [review]
Ah... this wasn't what I meant with my suggestion above.  My main goal was to avoid magic string constants.

A more straightforward approach might be to define new `CreatedFromTransform(ActorInfo fromInfo)` and `CreatedFromPlaceBuilding(bool lineBuild)` Init types that we add by default in these two places.  This then allows other actor creation traits to reuse these or add their own.

File: OpenRA.Mods.Common/Traits/Player/PlaceBuilding.cs [hunk 1]
```diff
@@ -16,6 +16,8 @@
 
 namespace OpenRA.Mods.Common.Traits
 {
+	public class PlaceBuildingInit : IActorInit { }
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Player/PlaceBuilding.cs]: Can you please add a comment explaining that this is used by third party mods to distinguish the build source?

**Review by pchote — DISMISSED** [review]
LGTM now.

**Review by pchote — APPROVED** [review]

**Review by GraionDilach — COMMENTED** [review]
:+1:

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/692b5a4ec3fcc0b86d5c8cc88214005e3aa1e4da...4bb35886589323348cdb613b04b10c72f0d7fc99)

**Pull request merged**